pennylane/math/utils.py

```diff
@@ -80,6 +80,28 @@ def _allclose_sparse(a, b, rtol=1e-05, atol=1e-08):
     return max_diff <= atol + rtol * max_b
 
 
+def _mixed_shape_match(a, b):
+    """Check if the shapes of two matrices of mixed types are compatible for comparison.
+
+    Args:
+        a, b: matrices to compare
+
+    Returns:
+        bool: True if the shapes are compatible
+    """
+    a_shapes = a.shape
+    b_shapes = b.shape
+    # Take the product, if inequal then false
+    if np.prod(a_shapes) != np.prod(b_shapes):
+        return False
+    # Make the sets of shapes, and ignore '1'
+    a_shape_set = set(i for i in a_shapes if i != 1)
+    b_shape_set = set(i for i in b_shapes if i != 1)

```

Comment:
I think it may be faster to first create the set then `discard` the 1 entry. Though, this isn't likely needed for the scale of problems run into, and may be beneficial for larger multidim tensor data.